Typescript compilation error regarding a bignumber.js import

#### Expected behavior

Build web3 with typescript without errors

#### Actual behavior

```
node_modules/web3-core/types/index.d.ts(31,23): error TS2307: Cannot find module 'bignumber.js'.
node_modules/web3-eth/types/index.d.ts(51,25): error TS2307: Cannot find module 'bignumber.js'.
```

see this Github Action to reproduce, logs, and configuration 
https://github.com/dapplion/web3-tsc-test/runs/461815145

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -151,3 +151,4 @@ Released with 1.0.0-beta.37 code base.
 ### Fixed
 
 - Add missing subscription.on('connected') TS type definition (#3319)
+- Add missing bignumber.js dependency for TS types (#3386)

--- a/packages/web3-core/package.json
+++ b/packages/web3-core/package.json
@@ -15,6 +15,7 @@
     "dependencies": {
         "@types/bn.js": "^4.11.4",
         "@types/node": "^12.6.1",
+        "bignumber.js": "^9.0.0",
         "web3-core-helpers": "1.2.6",
         "web3-core-method": "1.2.6",
         "web3-core-requestmanager": "1.2.6",

